vyper/builtin_functions/functions.py

```diff
@@ -1412,6 +1507,14 @@ def evaluate(self, node):
             value = (value << shift) % (2 ** 256)
         return vy_ast.Int.from_node(node, value=value)
 
+    def infer_arg_types(self, node):
```

Comment:
why override the default implementation?

Reply:
Currently, there is a disparity for shift - the input type is defined as `SignedIntegerAbstractType` but all the validation is based on `int128` (perhaps `int128` was the only signed type then?). Maybe the input type should be defined as `int128` instead?